### PR TITLE
Fix Issue #216: Removed Excess "Empty Comment" labels

### PR DESCRIPTION
**Pull request by cmata23x3**
This fix address issue #216 by removing the comments that would unnecessarily make "Empty Comment". Right now the server sends Javascript object in response to `getNotes`. `getNotes` appears to get all comments for a file for the user's permissions level. For admins, all comments are gotten, but for students, comments that are shared with "Instructors and TAs Only" are left out. The issue that causes the "Empty Comment" to be rendered are when an admin replies to the a "Instructors and TAs Only", but makes that comment shared with "The Entire Class". So now these comments are shown as "Empty Comments". 

This fix goes through and removes any comments that are not root comments and whose parent comment is not also in the comments returned. 

This fix is difficult to view because it is removing comments. Will discuss with @karger to determine how to best demo.

### CODE REVIEW

**Comment by sachazyto** [comment]
Looks good.  I would recommend to add that code as a component, to we can reuse it in those 3 places

**Comment by cmata23x3** [comment]
This fix now removes comments that are public replies to private messages. The idea is this would be a legacy fix. As it is, NB has already allowed public replies to private messages so this fix removes those from the view and prevents cluttering on the client side. 

The next step is to prevent this from happening is by repairing the permissions structure. This is an issue that has been opened up as #227.

**Comment by cmata23x3** [comment]
Updated the logic to be in `contents/modules/dev/pers.js` and the perspectives have been updated to use the new function. Also the function now also removes comments that are just empty strings (Legacy fix).

File: content/ui/admin/init.collage.js [hunk 1]
```diff
@@ -156,7 +156,7 @@ GLOB.pers.init = function(){
         });
     
     $.concierge.addComponents({
-        notes_loader:    function(P, cb){GLOB.pers.call("getNotes", P, cb);}, 
+        notes_loader: function(P, cb){GLOB.pers.__components.notes_loader(P, cb);},
```
Comment (sachazyto) [inline content/ui/admin/init.collage.js]: I think this line should be removed. Same about the other files: this is just assigning the object to itself.

Comment (karger) [inline content/ui/admin/init.collage.js]: christian says not---that GLOB.pers.__components.notes_loader is a *different* notes_loader function than the defined in concierge.addComponents.  GLOB.pers.__components.notes_loader is already defined to GLOB.pers.call("getNotes"), so having concierge's notes_loader invoke __components.notes_loader still invokes getNotes but now we've hidden away that implementation in pers.__components.

**Pull request closed without merging**